store/store_asserts.go

```diff
@@ -107,6 +99,71 @@ func (s *Store) Assertion(assertType *asserts.AssertionType, primaryKey []string
 	return asrt, nil
 }
 
+// SeqFormingAssertion retrieves the sequence-forming assertion for the given
+// type (currently validation-set only) and primary or sequence key.
+func (s *Store) SeqFormingAssertion(assertType *asserts.AssertionType, seqOrPrimaryKey []string, user *auth.UserState) (asserts.Assertion, error) {
```

Comment:
my preference would be to have a sequence int argument, that can be -1 or something to mean latest. This is mostly because most of the rest of the code will carry around sequence values as ints.

Reply:
Yes, that's fine, I was considering this but wasn't quite sure. Will change, thanks.